### PR TITLE
Support importing mods by dropping Zip or Folder

### PR DESCRIPTION
**Pull request by outercloudstudio**
Succeeds #35  

# Motivation
Improve quality of life and reduce user friction when importing mods.

# Screenshot
image

### CODE REVIEW

**Comment by FrederoxDev** [comment]
Thanks Outer! I'll take a look into these changes in a little bit!

**Comment by outercloudstudio** [comment]
By the way, the tsconfig target had to be changed to a newer es version to support iterating the items from the drop event.

File: src/components/DropWindow.tsx [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+		function saveFolderTo(
+			destination: string,
+			folder: FileSystemDirectoryEntry
+		) {
+			const reader = folder.createReader()
+
+			reader.readEntries((entries) => {
+				for (const entry of entries) {
+					if (entry.isFile) {
+						;(entry as FileSystemFileEntry).file((blob) => {
```
Comment (pappsamat) [inline src/components/DropWindow.tsx:112]: what is that semicolon lol

**Comment by FrederoxDev** [comment]
Cool, ok 👍 I don't think there was any reasons for us being on an older es version anyway

File: src/components/DropWindow.tsx (hunk 1 above)
Comment (outercloudstudio) [reply src/components/DropWindow.tsx:112]: Allows the type cast in parenthesis.

Comment (pappsamat) [reply src/components/DropWindow.tsx:112]: I didn't mean to start a review

**Comment by FrederoxDev** [comment]
image

I seem to get errors consistently when trying to import a .zipped mod

**Comment by FrederoxDev** [comment]
that folder doesn't exist before importing

**Comment by FrederoxDev** [comment]
Ah wait actually I realised it was because it was a zip with a folder inside, I think maybe some validation would be good here to check if its a valid mod by seeing if it has a mod.json data file

**Pull request merged**